packages/flutter/lib/src/material/ink_sparkle.dart

```diff
@@ -298,68 +311,70 @@ class InkSparkle extends InteractiveInkFeature {
-  Shader _createRippleShader() {
+  void _updateFragmentShader() {
     const double turbulenceScale = 1.5;
     final double turbulencePhase = _turbulenceSeed + _radiusScale.value;
     final double noisePhase = turbulencePhase;
     final double rotation1 = turbulencePhase * _rotateRight + 1.7 * math.pi;
     final double rotation2 = turbulencePhase * _rotateLeft + 2.0 * math.pi;
     final double rotation3 = turbulencePhase * _rotateRight + 2.75 * math.pi;
 
-    return _InkSparkleFactory._shaderManager!.shader(
-      // The following uniforms are the same throughout the animation.
-      uColor: _colorToVector4(_color),
-      uSparkleColor: Vector4(1.0, 1.0, 1.0, 1.0),
-      uBlur: 1.0,
-      uMaxRadius: _targetRadius,
-      uResolutionScale: Vector2(1.0 / _width, 1.0 / _height),
-      uNoiseScale: Vector2(_noiseDensity / _width, _noiseDensity / _height),
-
-      // The following uniforms update each frame of the animation.
-      uCenter: _center.value,
-      uRadiusScale: _radiusScale.value,
-      uAlpha: _alpha.value,
-      uSparkleAlpha: _sparkleAlpha.value,
-
-      // The following uniforms are driven by the turbulence phase and change
-      // each frame of the animation. These uniforms are uses to modify the
-      // default (if these fields are unset or 0) circular outer ring to a
-      // non-uniform shape that is more like an actual ink splash. In addition
-      // to the positional based triangular noise created in the shader, these
-      // uniforms also vary the appearance of the sparkles even when the same
-      // location is tapped.
-      uTurbulencePhase: turbulencePhase,
-      uNoisePhase: noisePhase / 1000.0,
-      uCircle1: Vector2(
-        turbulenceScale * 0.5 + (turbulencePhase * 0.01 * math.cos(turbulenceScale * 0.55)),
-        turbulenceScale * 0.5 + (turbulencePhase * 0.01 * math.sin(turbulenceScale * 0.55)),
-      ),
-      uCircle2: Vector2(
-        turbulenceScale * 0.2 + (turbulencePhase * -0.0066 * math.cos(turbulenceScale * 0.45)),
-        turbulenceScale * 0.2 + (turbulencePhase * -0.0066 * math.sin(turbulenceScale * 0.45)),
-      ),
-      uCircle3: Vector2(
-        turbulenceScale + (turbulencePhase * -0.0066 * math.cos(turbulenceScale * 0.35)),
-        turbulenceScale + (turbulencePhase * -0.0066 * math.sin(turbulenceScale * 0.35)),
-      ),
-      uRotation1: Vector2(math.cos(rotation1), math.sin(rotation1)),
-      uRotation2: Vector2(math.cos(rotation2), math.sin(rotation2)),
-      uRotation3: Vector2(math.cos(rotation3), math.sin(rotation3)),
-    );
-  }
-
-  Vector4 _colorToVector4(Color color) {
-    return Vector4(
-        color.red / 255.0,
-        color.blue / 255.0,
-        color.green / 255.0,
-        color.alpha / 255.0,
-      );
+    _fragmentShader
+      // uColor
+      ..setFloat(0, _color.red / 255.0)
+      ..setFloat(1, _color.green / 255.0)
```

Comment:
Fixed the order here.

Reply:
woops, nice!

Reply:
> Fixed the order here.

This is not fixed in 3.3.10(latest stable channel).
So when will this be fixed?